### ISSUE TITLE
QC for Drosophila sechellia

### ISSUE THREAD

**Issue by jradrion** [popsim-consortium/stdpopsim#873]
**PR for new species:** {https://github.com/popsim-consortium/stdpopsim/pull/872}

If you volunteer to QC this species, please use the checklist below.
While this list is intended to be comprehensive, it may not be exhaustive.
Where relevant, the QC reviewer should identify that parameter values match
those given in the linked citation(s).

- [ ] Recombination rate.
  - This might be genome-wide, or per-chromosome. Both are fine.
  - Check there's a comment describing where it came from, and/or how calculated.
  - From a publication? Check the value(s) match the publication.
  - Calculated somehow? Average over a recombination map? Redo the calculation.
- [ ] Mutation rate.
  - This might be genome-wide, or per-chromosome. Both are fine.
  - Check there's a comment describing where it came from, and/or how calculated.
  - From a publication? Check the value(s) match the publication.
  - Calculated somehow? Redo the calculation.
- [ ] Recombination map (if present).
  - Does it match the assembly? Liftover is fine, if clearly stated.
  - Is the description/long_description a good summary of how the map was created?
- [ ] Population size.
- [ ] Generation time.

For each citation, check:
- Doi link.
- Is publication a preprint? Is there a peer-reviewed publication instead?
- Is the year correct.
- Is the author correct (spelling, accents/ligatures/etc.)

Citations are required for:
- [ ] Genome reference assembly.
- [ ] Mutation rate.
- [ ] Recombination rate.
- [ ] Recombination map(s) (if relevant).
- [ ] Population size.
- [ ] Generation time.

**Comment by jeromekelleher** [comment]
Note that because the PR for this species used data from manually drawn NCBI, the test file is not stubbed out correctly and will need to be manually updated to include the correct chromosomes.

**Comment by vitorpavinato** [comment]
I can do this if anyone hasn't picked it yet.

**Comment by petrelharp** [comment]
Related: #885.

**Comment by petrelharp** [comment]
Hi, @vitorpavinato - @percyfal is around and willing to help if you haven't got to this yet?

**Comment by percyfal** [comment]
@vitorpavinato @petrelharp so I just redid the recombination rate calculations and now I recall why I ended up doing an independent liftover, to make sure I understood what was going on. I'll post the initial commands here to get started. I have a copy of the stdpopsim repo in ~/dev/stdpopsim and qc is done in ~/tmp/qc_DroSec. I'm working in a conda environment called `stdpopsim` with dependencies defined in [requirements/development.txt](https://github.com/popsim-consortium/stdpopsim/blob/main/requirements/development.txt). Btw, to run the liftover script the UCSC liftOver tool has to be installed; it isn't part of the stdpopsim development requirements but maybe should be?

```
conda activate stdpopsim
mamba install ucsc-liftover
mkdir -p ~/tmp/qc_DroSec
cd ~/tmp/qc_DroSec
python ~/dev/stdpopsim/maintenance/liftOver_comeron2012.py  --winLen 1000 --gapThresh 1000000 --useAdjacentAvg --retainIntermediates
```
The script will generate liftOver files, from which rates can be calculated. I recall doing this and not being able to reproduce the numbers in the catalog; then I added a print statement to the script [here (L284)](https://github.com/popsim-consortium/stdpopsim/blob/main/maintenance/liftOver.py#L284):
```
sys.stdout.write(f"RATE {chromName}: {new_chrom_average}\n")
sys.stdout.flush()
```
and reran the script like so (including print statements):
```
python ~/dev/stdpopsim/maintenance/liftOver_comeron2012.py  --winLen 1000 --gapThresh 1000000 --useAdjacentAvg --retainIntermediates 2>&1 | grep RATE
RATE chr2R: 2.6691562603887418
RATE chr3L: 1.795563724129453
RATE chr2L: 2.4048709303512266
RATE chr3R: 1.973177402673272
RATE chrX: 2.996015709213288
RATE chr3L: 1.7971860642049764
RATE chr2L: 2.4251899122807017
RATE chr3R: 1.973177402673272
RATE chrX: 2.999964904556056
RATE chr2R: 2.6691539884127593
```
Compare the values in the catalog:
```
    "2L": 2.40462600791e-08,
    "2R": 2.23458641776e-08,
    "3L": 1.79660308862e-08,
    "3R": 1.71642045777e-08,
    "4": 2.00579550709e-08,
    "X": 2.89650687913e-08,
```
In #885 it is pointed out that this can have to do with how telomeres are dealt with (https://github.com/popsim-consortium/stdpopsim/pull/885#issuecomment-823612056). The subsequent step would therefore be to look at the liftover files and use the rates, averaging as suggested in the comment above.

**Comment by percyfal** [comment]
And now I reread to the end of #885; @izabelcavassim did indeed suggest to use the average values from Cameron 2010 and not base the results on the liftover at all, IIUIC.

**Comment by vitorpavinato** [comment]
So, for the recombination rates, I should use the average values from Cameron 2010?

**Comment by petrelharp** [comment]
Yes - it's agreed that we should put in for the per-chromosome recombination rates the average values from Cameroon et al. You'll need to change that in the original species definition as well as in QC (and I'll verify).

**Comment by percyfal** [comment]
cf qc pull requests #885 and #1077

**Comment by vitorpavinato** [comment]
Almost done; I need to run the test and commit/push.
